pkg/diff/chunk.go

```diff
@@ -502,19 +391,22 @@ func getSplitFields(table *model.TableInfo, splitFields []string) ([]*model.Colu
 		splitCols = append(splitCols, col)
 	}
 
-	indexColumns := dbutil.FindAllColumnWithIndex(table)
+	if len(splitCols) != 0 {
+		return splitCols, nil
+	}
 
-	// user's config had higher priorities
-	for _, col := range append(append(splitCols, indexColumns...), table.Columns...) {
-		if _, ok := colsMap[col.Name.O]; ok {
-			continue
+	for _, col := range table.Columns {
+		colsMap[col.Name.O] = col
+	}
+	indices := dbutil.FindAllIndex(table)
```

Comment:
Do indices (returned from `FindAllIndex`) including some indices with some **ignored** columns?

Reply:
no, the table info already removed column and index with this column